addon/components/ember-table.js

```diff
@@ -252,41 +267,32 @@ export default class EmberTable2 extends Component {
   }
 
   /**
-   * There are cases where the sum of all column width is smaller than the container width. In this
-   * case, we want to auto increase width of some column. This function handles that logic.
-   * TODO(Billy): rewrite this function's logic to take into account min & max width.
+   * There are cases where the sum of all column width is smaller or bigger than the container
+   * width. In this case, we might want to adjust width of every single column.
    */
   fillupColumn() {
     const columns = this.get('columns');
     const tableWidth = this.get('_width');
     const sum = this.get('allColumnWidths');
+    const tableFillupMode = this.get('tableFillupMode');
 
-    if (sum < tableWidth - 1) {
-      let delta = tableWidth - sum - 1;
-      // If the table has fixed column, add all width difference to the fixed column. Otherwise,
-      // split the diff among all columns.
-      if (this.get('hasFixedColumn')) {
+    if (sum !== tableWidth) {
+      const delta = tableWidth - sum - 1;
+      // Distribute the delta in pixel among columns according to the table fill up mode.
+      if (tableFillupMode === TABLE_FILLUP_MODE_FIRST_COLUMN) {
         const [column] = columns;
         column.set('width', column.get('width') + delta);
-      } else {
-        // Split delta equally among columns.
-        let columnDelta = delta / columns.length;
-        if (columnDelta < 1) {
-          columnDelta = 1;
-        }
+      } else if (tableFillupMode === TABLE_FILLUP_MODE_EQUAL_COLUMN) {
+        // Split delta by their proportion.
+        const columnDelta = delta / columns.length;
```

Comment:
do we need to do something for the remainder?

Reply:
This is real number division. You do not have to worry about remainder.